src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -1416,6 +1416,126 @@ def _get_iot_central_app_by_name(client, app_name):
     return target_app
 
 
+def iot_central_list_private_link_resource(client, app_name=None, connection_id=None, resource_group_name=None, source_type=None):
+    if app_name and resource_group_name and source_type:
+        if source_type.lower() != 'microsoft.iotcentral/iotapps':
+            raise InvalidArgumentValueError(
+                "Type must be Microsoft.IoTCentral/iotApps")
+    elif connection_id:
+        id_list = connection_id.split('/')
+        if id_list[id_list.index('providers') + 1].lower() != 'microsoft.iotcentral':
+            raise InvalidArgumentValueError(
+                "Type must be Microsoft.IoTCentral/iotApps")
+        resource_group_name = id_list[id_list.index('resourceGroups') + 1]
+        app_name = id_list[id_list.index('iotApps') + 1]
+    else:
+        raise RequiredArgumentMissingError(
+            "Must provide id or application name, resource group, and type.")
```

Comment:
"Must provide private endpoint connection resource ID or resource name, resource group, and resource type."

We should be very clear with our error message so customer are able to take appropriate action.

FYI, "application name" is a term we use within the  Central team. Azure CLI is aimed at users who are familiar with Azure/ARM concepts. The "application name" is actually the "resource name"

Reply:
Nvm, is seems like "application name" is common naming for all of Azure CLi.